quantum/quantum.c

```diff
@@ -437,6 +437,16 @@ bool process_record_quantum(keyrecord_t *record) {
       return false;
       // break;
     }
+    case GRAVE_ESC: {
+      void (*method)(uint8_t) = (record->event.pressed) ? &add_key : &del_key;
+      uint8_t shifted = get_mods() & (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT));
+
+      if(layer_state == 0)
```

Comment:
why is this layer check needed? This seems to limit its usage to the layer condition exactly.